src/main/kotlin/io/snyk/plugin/ui/jcef/ThemeBasedStylingGenerator.kt

```diff
@@ -0,0 +1,119 @@
+package io.snyk.plugin.ui.jcef
+
+import com.intellij.openapi.editor.colors.ColorKey
+import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.editor.colors.EditorColorsScheme
+import com.intellij.openapi.editor.colors.FontPreferences
+import com.intellij.ui.jcef.JBCefBrowserBase
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.UIUtil
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+import java.awt.Color
+
+class ThemeBasedStylingGenerator {
+    fun toCssHex(color: Color): String {
+        return "#%02x%02x%02x".format(color.red, color.green, color.blue)
+    }
+
+    fun shift(
+        colorComponent: Int,
+        d: Double,
+    ): Int {
+        val n = (colorComponent * d).toInt()
+        return n.coerceIn(0, 255)
+    }
+
+    fun getCodeDiffColors(
+        baseColor: Color,
+        isHighContrast: Boolean,
+    ): Pair<Color, Color> {
+        val addedColor =
+            if (isHighContrast) {
+                Color(28, 68, 40) // high contrast green
+            } else {
+                Color(shift(baseColor.red, 0.75), baseColor.green, shift(baseColor.blue, 0.75))
+            }
+
+        val removedColor =
+            if (isHighContrast) {
+                Color(84, 36, 38) // high contrast red
+            } else {
+                Color(shift(baseColor.red, 1.25), shift(baseColor.green, 0.85), shift(baseColor.blue, 0.85))
+            }
+        return Pair(addedColor, removedColor)
+    }
+
+    fun generate(jbCefBrowser: JBCefBrowserBase): CefLoadHandlerAdapter {
+        val isDarkTheme = EditorColorsManager.getInstance().isDarkEditor
+        val isHighContrast =
+            EditorColorsManager.getInstance().globalScheme.name.contains("High contrast", ignoreCase = true)
```

Comment:
Wondering: should we also set the other font? It looks like the font is different in "native" and "html", and that maybe is due to not taking the theme font?

Reply:
We could, but what would be the point of it? Is it to make the change more gradual once we enable this feature?